### ISSUE TITLE
It is hard to extend ol functionality

### ISSUE THREAD

**Issue by sookoll** [openlayers/openlayers#9038]
I need that GPX format parses some custom nodes in wpt. At first I thought that it will be easy task, as I have extent ol classes before. Just extend ol/format/GPX class and if I'm lucky, rewrite one method. But when I looked at how new modules are built up, I realize, that You have made big regression on that. Most of the code are left out of class prototype.
So to achive that one wpt custom node is parsed, I end up nearly 400 lines of duplicated code. Like wtf?

**Comment by jalik** [comment]
@sookoll do you want to use functions that are not exported in this file?
https://github.com/openlayers/openlayers/blob/master/src/ol/format/GPX.js

**Comment by sookoll** [comment]
That's correct. In order to add new xml node as feature property, I needed to make changes in readWpt function, which is not exported. I created my own. And to redefine, which function are used, I had to overwrite XPX.prototype.readFeaturesFromNode, then copy-paste most of the not exported constants and functions, as they are needed by readFeaturesFromNode. For example, I needed just modified readWpt, but since readFeaturesFromNode use GPX_PARSERS constant, and I don't want to loose track and route functionality, I had to add also track and route reader functions + all dependant code.
It is general problem. How to let users to extend modules without trouble. Current approach do not support that.
For my particular need, there could be more general solution for reading custom nodes. For example, instead this hard coded extensions parsing, perhaps let users to input, what custom nodes they want to parsed to feature properties addition to default ones.
```
import GPX from 'ol/format/GPX'
const format = new GPX({
  customNodes: ['extensions', 'groundspeak:cache']
})
```
